### ISSUE TITLE
Errors when loading docs on staging

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#1739]
<img width="1424" alt="CleanShot 2022-05-18 at 08 20 18@2x" src="https://user-images.githubusercontent.com/65520/168970836-57a18b0c-f38d-4ea7-8029-734bef13d7d1.png">

This looks like an issue introduced by Cloudflare, perhaps as part of https or compression in general.

When loading the docs via the IP address or http://spi-dev.swiftpackageindex.com it works:

- http://40.87.31.205/SwiftPackageIndex/SemanticVersion/main/documentation/semanticversion
- http://spi-dev.swiftpackageindex.com/SwiftPackageIndex/SemanticVersion/main/documentation/semanticversion

<img width="1299" alt="CleanShot_2022-05-18_at_07 58 162x" src="https://user-images.githubusercontent.com/65520/168971196-2f5456f5-e754-4027-9e3a-b429de6fbe42.png">

**Comment by finestructure** [comment]
(we should fix that favicon.ico 404)

**Comment by finestructure** [comment]
Setting the [`cache-control: no-transform`](a8537a40531392bf51cdcfb42268b8b8ccf5f642) header for the documentation routes didn't take for some reason:

```
❯ curl -Is https://staging.swiftpackageindex.com/SwiftPackageIndex/SemanticVersion/main/js/chunk-vendors.00bf82af.js | grep cache-control
cache-control: max-age=14400
```

```
❯ curl -s https://staging.swiftpackageindex.com/api/version
{"version":"a8537a40531392bf51cdcfb42268b8b8ccf5f642"}
```

**Comment by finestructure** [comment]
Looking at the headers we've got

```
❯ curl -Is https://staging.swiftpackageindex.com/SwiftPackageIndex/SemanticVersion/main/js/chunk-vendors.00bf82af.js
HTTP/2 200
date: Wed, 18 May 2022 06:53:54 GMT
content-type: application/javascript
content-length: 97294
x-amz-id-2: x+2YccxdCFCw8i/JP5z3e5UksAODrTRuq3FcYmMy4YqWJ8qbxoXromV49G9zdEUjjlR9b0wj23E=
x-amz-request-id: JSKEV8ZHW4KZ5QPY
content-encoding:
last-modified: Mon, 16 May 2022 17:10:05 GMT
etag: "99f6071582172fc5ee9df8864ce86089"
host: staging.swiftpackageindex.com
accept-encoding: gzip
cf-ipcountry: FR
x-forwarded-for: 37.167.58.28
x-forwarded-proto: https
cf-visitor: {"scheme":"https"}
accept: */*
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.5 Safari/605.1.15
accept-language: en-GB,en;q=0.9
referer: https://staging.swiftpackageindex.com/SwiftPackageIndex/SemanticVersion/main/documentation/semanticversion
cf-connecting-ip: 37.167.58.28
cdn-loop: cloudflare
cache-control: max-age=14400
cf-cache-status: HIT
age: 3462
accept-ranges: bytes
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=47y3LtPc3dq5ebWibkRkavqsb3zupGvTz6e2RwQ5MgqCA3Su5AiAzqWLfwpS%2BG%2FBkV3cWVTx2y0aqAYeD3KYvVIO4aPoFW6Q%2Bs0q2L76S7r2pJKgOYubxURlCDpjwiAT0BeaKycfMtohVsMqaXGV"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 70d2adb1da84eddb-CDG
```

and

```
❯ curl -Is http://40.87.31.205/SwiftPackageIndex/SemanticVersion/main/js/chunk-vendors.00bf82af.js
HTTP/1.1 200 OK
x-amz-id-2: QjeDY+4UZL1vB4a6eDW0qS7a+a5hdh4llQvjI+CxctmXAzRXX+i39pxPzxuxvqUECOucgzfb3hI=
x-amz-request-id: GYP53GTZ5TRVSR6P
Date: Wed, 18 May 2022 06:52:34 GMT
Content-Encoding:
Last-Modified: Mon, 16 May 2022 17:10:05 GMT
ETag: "99f6071582172fc5ee9df8864ce86089"
Server: AmazonS3
Content-Length: 97294
Host: 40.87.31.205
User-Agent: curl/7.79.1
Accept: */*
content-type: application/javascript
cache-control: no-transform
connection: keep-alive
date: Wed, 18 May 2022 06:52:33 GMT
```

**Comment by finestructure** [comment]
And comparing `main.js`:

```
❯ curl -Is https://staging.swiftpackageindex.com/main.js
HTTP/2 200
date: Wed, 18 May 2022 06:54:54 GMT
content-type: application/javascript
content-length: 105325
etag: 1652855275.0-105325
cache-control: max-age=14400
cf-cache-status: MISS
accept-ranges: bytes
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=aFoHMHI6vCRGoc%2BKCS2P5O%2BONFjw3D7eQUd6%2FXRvnxkbWM3sjQ8Dy74%2FPUuiSTJzAd6GFJI2d5Ce4tzJUSDKEdDY4T0lMnQRmogVR%2BvZx2%2FKfsLXKOXnEhg0UP9%2BjkP8B9iscm1yO8nGm2FiLfve"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 70d2af282a913a99-CDG
```

```
❯ curl -Is http://40.87.31.205/main.js
HTTP/1.1 200 OK
etag: 1652855275.0-105325
content-type: application/javascript
content-length: 105325
connection: keep-alive
date: Wed, 18 May 2022 06:55:47 GMT
```

**Comment by finestructure** [comment]
I wonder if that empty `Content-Encoding` header is a problem.

**Comment by finestructure** [comment]
Potential fix #1740 - didn't think it fixed it due to CF cache not being purged and having 4h max age.

**Comment by finestructure** [comment]
That fix worked.
